### DISCUSSION TITLE
How to uninstall python and homebrew completely ?

### DISCUSSION THREAD

**Discussion by njacksonnie** [Homebrew/discussions#1823] [Getting started] [answered]
hi, I am on MacBook Pro M1. I've installed homebrew and python for work. and now I need to uninstall homebrew and python completely from my mac. I don't want to leave any type of leftovers in mac.

Didn't found anything on this particular topic. So any help would be highly appreciated. Thanks 😊 
and also I don't know any of those command to uninstall properly.

**Comment by SMillerDev** [comment, marked as answer]
- `brew uninstall python`
- https://docs.brew.sh/Installation#uninstallation

**Reply by njacksonnie** [reply]
hello thank you sir! i will follow your instructions.

and also I forgot to mention that I've installed cask in Homebrew. so how do I remove case too? or uninstalling home brew will remove cask completely too?

**Reply by SMillerDev** [reply]
You can remove anything you installed using the `brew uninstall` command.

**Reply by njacksonnie** [reply]
yes sir i understood your statement but also I want to confirm that, if I uninstall Homebrew without uninstalling cask, will it also uninstall cask with homebrew? or If I install homebrew in future, **cask will be there**?

am so sorry, am very new to this (just a month with first Mac). thank you for your patience 🙂

**Reply by SMillerDev** [reply]
I believe casks will remain after a homebrew uninstall.
